api/SubmitJob/cloudformation.yml

```diff
@@ -0,0 +1,64 @@
+AWSTemplateFormatVersion: 2010-09-09
+
+Outputs:
+
+  LambdaArn:
+    Value: !GetAtt Lambda.Arn
+
+Parameters:
+
+  JobQueueArn:
+    Type: String
+
+  JobDefinition:
+    Type: String
+
+Resources:
+
+  LogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub "/aws/lambda/${AWS::StackName}"
+      RetentionInDays: 90
+
+  Role:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Ref AWS::StackName
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          Action: sts:AssumeRole
+          Principal:
+            Service: lambda.amazonaws.com
+          Effect: Allow
+      Policies:
+      - PolicyName: policy
+        PolicyDocument:
+          Version: 2012-10-17
+          Statement:
+          - Effect: Allow
+            Action:
+            - logs:CreateLogStream
+            - logs:PutLogEvents
+            Resource: !GetAtt LogGroup.Arn
+          - Effect: Allow
+            Action: batch:SubmitJob
+            Resource:
+            - !Ref JobQueueArn
+            - !Ref JobDefinition
+
+  Lambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      FunctionName: !Ref AWS::StackName
+      Environment:
+        Variables:
+          JobQueue: !Ref JobQueueArn
+          JobDefinition: !Ref JobDefinition
+      Code: src/
```

Comment:
for trivial functions this can point directly at a `.py` file and eliminate the need for the directory, `Code: main.py`, but we'll need a directory back as soon as we need any python packages, e.g. `pip install -r requirements.txt -t src/`

Reply:
I think I like this because it allows us the ability to easily break the function into multiple files if it starts getting large